### PR TITLE
Move loading of PK-Sim libs to package init

### LINKED ISSUE #1433: Load PK-Sim DLL upon package loading
I think once PK-SIM.R can be loaded in Linux, it should be done in `.initPackage` and not as a separate function `initPKSim`.

- Load PKSim.R.dll in initPackage
- Remove function `initPKSim`
- Remove `ospsuiteEnv$isPKSimLoaded`
- Remove everything in `utilities-pksim`

### PR DESCRIPTION
**Pull request by PavelBal**
Fixes #1433

### CODE REVIEW

**Review by msevestre — COMMENTED** [review]
so cool

**Comment by Felixmil** [comment]
Doesn't it slow down package loading ?

**Comment by PavelBal** [comment]
> Doesn't it slow down package loading ?

It does indeed, but we just assume it should always be available?

**Comment by msevestre** [comment]
Well the initOnce can only be called when one really use PKSim stuff no?

File: R/init-package.R [hunk 1]
```diff
@@ -35,4 +35,9 @@
   rSharp::callStatic("OSPSuite.R.Api", "InitializeOnce", apiConfig)
 
   .initializeDimensionAndUnitLists()
+
+  # Load PK-Sim library
+  pksimR <- system.file("lib", "PKSim.R.dll", package = ospsuiteEnv$packageName)
+  rSharp::loadAssembly(pksimR)
+  rSharp::callStatic("PKSim.R.Api", "InitializeOnce")
```
Comment (msevestre) [inline R/init-package.R:42]: I am wondering if this should be done like it was before. Whast's the gain of ALWAYS loading PKSim?

Comment (PavelBal) [reply R/init-package.R:42]: Okay I can revert.

**Comment by Felixmil** [comment]
I'm in favor of loading PKSIM only when/if used.

**Comment by PavelBal** [comment]
Yepp, closing.

**Pull request closed without merging**